### PR TITLE
DEV: IMAP debugging improvements

### PR DESCRIPTION
**Pull request by martin-brennan**
Improvements to make console access to IncomingEmail more pleasant, and stopping certain IMAP logs from landing in the DB because they just create too much noise,

### CODE REVIEW

File: app/models/imap_sync_log.rb [hunk 1]
```diff
@@ -9,20 +9,26 @@ def self.levels
     @levels ||= Enum.new(:debug, :info, :warn, :error)
   end
 
-  def self.log(message, level, group_id = nil)
+  def self.log(message, level, group_id = nil, db = true)
```
Comment (lis2) [inline app/models/imap_sync_log.rb:12]: With arity of 4 I would consider named parameters. Don't have a strong opinion here so completely up to you

Comment (martin-brennan) [reply app/models/imap_sync_log.rb:12]: Fair call but this method is kind of like a more custom/internal method anyway so will leave as is

File: app/models/imap_sync_log.rb [hunk 2]
```diff
@@ -9,20 +9,26 @@ def self.levels
...
     now = Time.now.strftime("%Y-%m-%d %H:%M:%S.%L")
-    new_log = create(message: message, level: ImapSyncLog.levels[level], group_id: group_id)
+    new_log = nil
+
+    if db
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/models/imap_sync_log.rb:16]: Yup, will change

File: app/models/post.rb [hunk 3]
```diff
@@ -166,6 +166,10 @@ def self.find_by_detail(key, value)
     includes(:post_details).find_by(post_details: { key: key, value: value })
   end
 
+  def self.find_by_number(topic_id, post_number)
+    find_by(topic_id: topic_id, post_number: post_number)
+  end
+
```
Comment (lis2) [inline app/models/post.rb:172]: Is that helper used anywhere?

**Review by lis2 — APPROVED** [review]

File: app/models/post.rb (hunk 3 above)
Comment (martin-brennan) [reply app/models/post.rb:172]: No, neither are the other ones introduced here, they are all intended to make debugging this IMAP/email stuff way easier in `rails console`. This one for example turns `Topic.find(x).posts.find_by(post_number: y)` into `Post.find_by_number(x, y)`

**Pull request merged**